src/main/java/io/mosip/mimoto/controller/CredentialsController.java

```diff
@@ -0,0 +1,94 @@
+import io.mosip.mimoto.dto.mimoto.CredentialsSupportedResponse;
+import io.mosip.mimoto.dto.mimoto.VCCredentialRequest;
+import io.mosip.mimoto.dto.mimoto.VCCredentialResponse;
+import io.mosip.mimoto.exception.ApiNotAccessibleException;
+import io.mosip.mimoto.exception.IdpException;
+import io.mosip.mimoto.service.CredentialService;
+import io.mosip.mimoto.service.IdpService;
+import io.mosip.mimoto.service.IssuersService;
+import io.mosip.mimoto.util.DateUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.core.io.InputStreamResource;
+import org.springframework.http.*;
+import org.springframework.util.MultiValueMap;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.web.client.RestTemplate;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+
+import static io.mosip.mimoto.exception.PlatformErrorMessages.API_NOT_ACCESSIBLE_EXCEPTION;
+import static io.mosip.mimoto.exception.PlatformErrorMessages.MIMOTO_PDF_SIGN_EXCEPTION;
+
+@RestController
+@RequestMapping(value="/credentials")
+public class CredentialsController {
+
+    private static final String ID = "mosip.mimoto.credentials";
+    private final Logger logger = LoggerFactory.getLogger(CredentialsController.class);
+
+    @Autowired
+    IssuersService issuersService;
+
+    @Autowired
+    CredentialService credentialService;
+
+    @Autowired
+    IdpService idpService;
+
+    @PostMapping("/download")
+    public ResponseEntity<?> downloadCredentialAsPDF(
+            @RequestParam Map<String, String> params) {
+
+        ResponseWrapper<Object> responseWrapper = new ResponseWrapper<>();
+        responseWrapper.setId(ID);
+        responseWrapper.setVersion("v1");
+        responseWrapper.setResponsetime(DateUtils.getRequestTimeString());
+        try{
+
+            String issuerId = params.get("issuer");
+            String credentialType = params.get("credential");
+
+            logger.info("Initiated Token Call");
+            RestTemplate restTemplate = new RestTemplate();
+            IssuerDTO issuerDTO = issuersService.getIssuerConfig(issuerId);
+            HttpEntity<MultiValueMap<String, String>> request = idpService.constructGetTokenRequest(params, issuerDTO);
+            TokenResponseDTO response = restTemplate.postForObject(idpService.getTokenEndpoint(issuerDTO), request, TokenResponseDTO.class);
+            if(response == null) {
+                throw new IdpException("Exception occurred while performing the authorization");
+            }
+            logger.info("Initiated Download Credential Call");
+            IssuerDTO issuerConfig = issuersService.getIssuerConfig(issuerId);
+            CredentialIssuerWellKnownResponse credentialIssuerWellKnownResponse = credentialService.getCredentialIssuerWellknown(issuerId, credentialType);
```

Comment:
Line no 73 to 77 can be delegated to service as these are mandatory step to download credential

Reply:
Done.